### PR TITLE
[KYUUBI #2550] Fix swagger does not show the request/response schema issue

### LINKED ISSUE #2550: [Bug] Kyuubi swagger does not show the request/response schema
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

Kyuubi swagger does not show the request/response schema.

image
image

### Affects Version(s)

1.6.0

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

To close #2550

Refer the comments: https://github.com/apache/incubator-kyuubi/pull/1658#issuecomment-1003244581

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image

image

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
But for the response, it does not show the schema.
image

**Comment by cfmcgrady** [comment]
> But for the response, it does not show the schema.
> 
> cc @cfmcgrady , is it the same with before?

not sure, the schema should be shown.

**Comment by cfmcgrady** [comment]
Hi @turboFei 

The response schema is missing before. After setting response schema annotation, the OpenAPI works fine.

here is an example:

# GET /v1/sessions/execPool/statistic

```scala
  @ApiResponse(
    responseCode = "200",
    content = Array(new Content(
      mediaType = MediaType.APPLICATION_JSON,
      schema = new Schema(implementation = classOf[ExecPoolStatistic]))),
    description = "Get statistic info of background executors")
  @GET
  @Path("execPool/statistic")
  def execPoolStatistic(): ExecPoolStatistic = {
    ExecPoolStatistic(
      sessionManager.getExecPoolSize,
      sessionManager.getActiveCount)
  }
```

```json
    "/v1/sessions/execPool/statistic" : {
      "get" : {
        "tags" : [ "Session" ],
        "operationId" : "execPoolStatistic",
        "responses" : {
          "200" : {
            "description" : "Get statistic info of background executors",
            "content" : {
              "application/json" : {
                "schema" : {
                  "$ref" : "#/components/schemas/ExecPoolStatistic"
                }
              }
            }
          }
        }
      }
    }
```
截屏2022-05-05 下午1 57 43

**Comment by turboFei** [comment]
I think we can follow it in another pr. cc @yaooqinn

**Comment by turboFei** [comment]
> ```scala
>   def execPoolStatistic(): ExecPoolStatistic = {
> ```

Thanks.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala [hunk 1]
```diff
@@ -44,7 +44,8 @@ private[v1] class SessionsResource extends ApiRequestContext with Logging {
   @ApiResponse(
     responseCode = "200",
     content = Array(new Content(
-      mediaType = MediaType.APPLICATION_JSON)),
+      mediaType = MediaType.APPLICATION_JSON,
+      schema = new Schema(implementation = classOf[Seq[SessionData]]))),
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala:48]: generic type erasure: 
```
schema = new Schema(implementation = classOf[Seq[_]])
```

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala:48]: is there a better way for this?

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala:48]: will try
```
      schema = new ArraySchema(new Schema(implementation = classOf[Seq[SessionData]])))),

```

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala:48]:
```
    "/v1/sessions" : {
      "get" : {
        "tags" : [ "Session" ],
        "operationId" : "sessions",
        "responses" : {
          "200" : {
            "description" : "get the list of all live sessions",
            "content" : {
              "application/json" : {
                "schema" : {
                  "type" : "array",
                  "items" : {
                    "$ref" : "#/components/schemas/SessionData"
                  }
                }
              }
            }
          }
        }
      },
```

**Comment by turboFei** [comment]
It works fine now.

**Review by cfmcgrady — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**